### PR TITLE
Make javascript search work in more contexts

### PR DESCRIPTION
**Pull request by silviot**
The meaning of `\w` changed from python 2 to python 3.

> $ python3.5 -c "import re; print(re.compile(r'[a-zA-Z]\w*$').match(u'a\xe8'));"
> <_sre.SRE_Match object; span=(0, 2), match='aè'>
> 
> $ python -c "import re; print(re.compile(r'[a-zA-Z]\w*$').match(u'a\xe8'));"
> None

but the definition of what's an acceptable javascript identifier should not vary from case to case.

### CODE REVIEW

File: tests/test_jsdump.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# -*- coding: utf-8 -*-
```
Comment (tk0miya) [inline tests/test_jsdump.py]: Could you rename this to `test_util_jsdump.py`?

**Comment by tk0miya** [comment]
LGTM with nits

**Comment by tk0miya** [comment]
Is this related with any bugs? If true, it would be better to move to stable branch.

**Comment by silviot** [comment]
It is a bug on python > 3.

I had [a run of the test](https://travis-ci.org/silviot/sphinx/builds/143276911) without the fix to confirm this.

Should I open another pull request against the `stable` branch?

**Comment by tk0miya** [comment]
Ah, I meant that is there any undesirable behavior of HTML search related with this.
I want to merge this if this is a bug of the search feature.

**Comment by silviot** [comment]
@tk0miya Without this patch, the search index can be invalid javascript on python3.
For me this patch makes the search work, so I consider it a bug of the search feature.

**Comment by tk0miya** [comment]
Okay. Could you rebase this onto stable branch?

**Comment by tk0miya** [comment]
it would be nice if you add a CHANGES entry. Thanks,

**Comment by silviot** [comment]
Opened https://github.com/sphinx-doc/sphinx/pull/2766 with entry in CHANGES added. I squashed everything into a single commit.

**Pull request closed without merging**